android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/RegistrationServiceImpl.java

```diff
@@ -476,6 +494,83 @@ private List<Map<String, String>> getLabelValueDTOListString(Map<String, String>
         return labelValueMap;
     }
 
+    /**
+     * Validate machine location against registration center
+     * @throws Exception if location is outside allowed distance
+     */
+    private void validateLocation() throws Exception {
+        try {
+            GeoLocationDto geoLocation = this.registrationDto.getGeoLocationDto();
+            if (geoLocation == null) {
+                Log.w(TAG, "Geo location not available, skipping validation");
+                return;
+            }
+
+            // Get center coordinates
+            CenterMachineDto centerMachineDto = masterDataService.getRegistrationCenterMachineDetails();
+            if (centerMachineDto == null) {
+                Log.w(TAG, "Center details not found, skipping distance validation");
+                return;
+            }
+
+            List<RegistrationCenter> centers = registrationCenterRepository.getRegistrationCenter(
+                centerMachineDto.getCenterId());
+
+            if (centers == null || centers.isEmpty()) {
+                Log.w(TAG, "Center not found, skipping distance validation");
+                return;
+            }
+
+            RegistrationCenter center = centers.get(0);
+            String centerLatStr = center.getLatitude();
+            String centerLonStr = center.getLongitude();
+
+            if (centerLatStr == null || centerLonStr == null ||
+                centerLatStr.isEmpty() || centerLonStr.isEmpty()) {
+                Log.w(TAG, "Center coordinates not available, skipping distance validation");
+                return;
+            }
+
+            try {
+                double centerLatitude = Double.parseDouble(centerLatStr);
+                double centerLongitude = Double.parseDouble(centerLonStr);
+
+                // Calculate distance
+                double distance = locationValidationService.getDistance(
+                    geoLocation.getLongitude(), geoLocation.getLatitude(),
+                    centerLongitude, centerLatitude);
+
+                // Get max allowed distance from config
+                String maxDistanceStr = globalParamRepository.getCachedStringMachineToCenterDistance();
+                if (maxDistanceStr == null || maxDistanceStr.isEmpty()) {
+                    Log.e(TAG, "Max allowed distance configuration not found");
+                    throw new ClientCheckedException(context, R.string.err_004);
+                }
+
+                double maxAllowedDistance = Double.parseDouble(maxDistanceStr);
+
+                // Validate distance
+                if (distance > maxAllowedDistance) {
+                    String errorMsg = String.format("Machine is %.2f km from center (Max allowed: %.2f km)",
```

Comment:
How is this  message multi-lang compatible? this should be added in the language resource bundle